### DISCUSSION TITLE
Validity of certificate

### DISCUSSION THREAD

**Discussion by ashokvadivelu** [outfoxx/Shield#40] [General]
How to access the validity/expiration date of certificate like issuerName and subjectName

**Comment by kdubb** [comment]
When you have an instance of `Certificate`. You can use the `tbsCertificate` property to access all of the available certificate properties.

Example:
```swift
let cert: Certificate = retrieveCertificateFromSomewhere()

print(cert.tbsCertificate.issuer)
print(cert.tbsCertificate.subject)
print(cert.tbsCertificate.validity.notAfter)
```

**Reply by kdubb** [reply]
FYI, the property names come directly from the RFC for signed certificates.
